hera_sim/components.py

```diff
@@ -1,14 +1,18 @@
 """A module providing discoverability features for hera_sim.
 """
-
-import functools
+from __future__ import annotations
```

Comment:
This is not a Py2 thing -- it's something that I think was put in 3.8 or 3.9 and backported to 3.7+. It does make 3.6 break, but I'd argue that no-ones still using that. 

What this in particular does is means you can use type annotations referring to a given class name _inside_ that class. Without it, it would say that the class is not yet defined. I can just take out these annotations, but I don't want to :-)

Reply:
If you think 3.6 is no longer being used, then I think it's okay to leave this in--it sounds like a nice feature. Though we'll need to drop 3.6 testing.